api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -134,56 +139,93 @@ def _handle_incremental_upserts(self, existing_opportunity_ids: set[int]) -> Non
             .first()
         )
 
-        # Fetch opportunities that need processing from the queue
-        query = (
-            select(Opportunity)
-            .join(OpportunityChangeAudit)
-            .join(CurrentOpportunitySummary)
-            .where(
-                Opportunity.is_draft.is_(False),
-                CurrentOpportunitySummary.opportunity_status.isnot(None),
+        last_processed_opportunity: Opportunity | None = None
+        while True:
```

Comment:
There is a lot going on in this single loop, could we move some of the implementation into functions so it's easier to parse? Maybe something like a function for building the query?